Jenkinsfile

```diff
@@ -39,12 +94,19 @@ podTemplate(cloud: 'openshift', label: 'coreos-assembler', yaml: pod, defaultCon
             rm -rf src/config
 
             # in the future, the stream will dictate the branch in the prod path
-            coreos-assembler init --force https://github.com/coreos/fedora-coreos-config
+            coreos-assembler init --force --branch ${src_config_ref} ${src_config_url}
             """)
         }
 
         stage('Fetch') {
-            if (!devel) {
+            // XXX: drop `!prod && ` once we've uploaded prod builds there
+            if (!prod && s3_builddir) {
+                utils.shwrap("""
+                coreos-assembler buildprep s3://${s3_builddir}
+                """)
+            }
```

Comment:
i don't want to address this in this PR (we're very close to merging) but would like to brainstorm on how we make this step non-fatal in cases where we are starting with a fresh s3 bucket (i.e. someone set up a new s3 bucket and just brought this up in openshift for the first time).

Reply:
Hmm yeah, `buildprep` has:

```python
    builds = fetcher.fetch_json('builds.json')
    if len(builds) == 0:
        print("Remote has no builds!")
        return
```

though that still assumes that `builds.json` is present. Just need to tweak that.

IOW, I think this is something we should fix in cosa, not necessarily here.

Reply:
> IOW, I think this is something we should fix in cosa, not necessarily here.

we'd probably have to handle different errors differently, though? in one case builds.json doesn't exist and it's the first build and we want to go on. In the other case builds.json does exist but there was an error retrieving it (network issue, permissions issue, etc).

Reply:
Yeah, I think we should be able to key off of the response code in both cases. Will look into this.

Reply:
opened https://github.com/coreos/coreos-assembler/issues/543 for this